spark-jobs/src/main/scala/filodb/downsampler/DownsamplerUtil.scala

```diff
@@ -0,0 +1,34 @@
+package filodb.downsampler
+
+import scala.collection.mutable
+
+import filodb.core.binaryrecord2.RecordBuilder
+import filodb.core.metadata.Schema
+import filodb.downsampler.index.DSIndexJob.schemas
+import filodb.memory.{BinaryRegionLarge, MemFactory}
+import filodb.memory.format.{UnsafeUtils, ZeroCopyUTF8String}
+import filodb.memory.format.ZeroCopyUTF8String._
+
+object DownsamplerUtil {
+
+  /**
+    * create downsampled Partkey from source/raw Partkey
+    */
+  def downsampledPk(partKey: Array[Byte], dsSchema: Schema, offHeapMem: MemFactory): Array[Byte] = {
+    val pkStrPairs = schemas.part.binSchema.toStringPairs(partKey, UnsafeUtils.arayOffset)
+    downsampledPk(pkStrPairs, dsSchema, offHeapMem)
+  }
+
+  /**
+    * create downsampled Partkey from the source/raw Partkey
+    */
+  def downsampledPk(pkStrPairs: Seq[(String, String)], dsSchema: Schema, offHeapMem: MemFactory): Array[Byte] = {
+    val dsRecordBuilder = new RecordBuilder(offHeapMem)
```

Comment:
Lets reuse RecordBuilder

Reply:
modified the code to use different strategy